hr_period/models/hr_fiscal_year.py

```diff
@@ -0,0 +1,299 @@
+# Copyright 2015 Savoir-faire Linux. All Rights Reserved.
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.tools import DEFAULT_SERVER_DATE_FORMAT as DF
+from odoo.exceptions import UserError
+from dateutil.relativedelta import relativedelta
+from dateutil.rrule import (YEARLY,
+                            MONTHLY,
+                            WEEKLY,
+                            DAILY)
+from datetime import datetime
+
+strptime = datetime.strptime
+strftime = datetime.strftime
+
+INTERVALS = {
+    'annually': (relativedelta(months=12), 1),
+    'semi-annually': (relativedelta(months=6), 2),
+    'quarterly': (relativedelta(months=3), 4),
+    'bi-monthly': (relativedelta(months=2), 6),
+    'semi-monthly': (relativedelta(weeks=2), 24),
+    'monthly': (relativedelta(months=1), 12),
+    'bi-weekly': (relativedelta(weeks=2), 26),
+    'weekly': (relativedelta(weeks=1), 52),
+    'daily': (relativedelta(days=1), 365),
+}
+
+
+@api.model
+def get_schedules(self):
+    return [
+        ('annually', _('Annually (1)')),
+        ('semi-annually', _('Semi-annually (2)')),
+        ('quarterly', _('Quarterly (4)')),
+        ('bi-monthly', _('Bi-monthly (6)')),
+        ('monthly', _('Monthly (12)')),
+        ('semi-monthly', _('Semi-monthly (24)')),
+        ('bi-weekly', _('Bi-weekly (26)')),
+        ('weekly', _('Weekly (52)')),
+        ('daily', _('Daily (365)')),
+    ]
+
+
+@api.model
+def get_payment_days(self):
+    expr = _('%s day of the next period')
+    expr_2 = _('%s day of the current period')
+    return [
+        ('1', expr % _('First')),
+        ('2', expr % _('Second')),
+        ('3', expr % _('Third')),
+        ('4', expr % _('Fourth')),
+        ('5', expr % _('Fifth')),
+        ('0', expr_2 % _('Last')),
+    ]
+
+
+class HrFiscalYear(models.Model):
+    _name = 'hr.fiscalyear'
+    _inherit = 'date.range'
+    _description = 'HR Fiscal Year'
+
+    @api.model
+    def _default_date_start(self):
+        today = datetime.now()
+        return datetime(today.year, 1, 1).strftime(DF)
+
+    @api.model
+    def _default_date_end(self):
+        today = datetime.now()
+        return datetime(today.year, 12, 31).strftime(DF)
+
+    @api.model
+    def _default_type(self, company_id=False):
```

Comment:
@mreficent would it be possible to test-cover this method?

Reply:
Odd, it's actually called from https://github.com/OCA/hr/pull/524/files#diff-a191dafb7bdefc26785687b9040c0715R246